### PR TITLE
Accept CloudEvents in any order for CE reconcile test

### PR DESCRIPTION
**Pull request by sbwsg**
# Changes

Fixes https://github.com/tektoncd/pipeline/issues/2992

Prior to this commit test cloud events received during reconcile were
required to be emitted in a strict order. We've been seeing some flakes in
our unit tests related to this and in
https://github.com/tektoncd/pipeline/issues/2992 we posited that this might
be happening because the cloud events controller emits each event in a
separate go routine. We added some extra logging to try and catch the
details of the flakes and it does appear that the events are being sent
out-of-order.

This commit adds a new helper func that allows the expected cloudevents
to appear in any order while still failing the test if an unexpected event
(or not enough events) are received.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Release notes block has been filled in or deleted (only if no user facing changes)

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/taskrun/taskrun_test.go [hunk 1]
```diff
@@ -350,6 +353,47 @@ func eventFromChannel(c chan string, testName string, wantEvents []string) error
...
+					expected = expected[:len(expected)-1]
+					break
+				}
+			}
+			if !found {
+				return fmt.Errorf("unexpected event received: %q", event)
+			}
+		case <-timer.C:
+			if len(expected) != 0 {
+				return fmt.Errorf("timed out waiting for %d more events", len(expected))
```
Comment (GregDritschler) [inline pkg/reconciler/taskrun/taskrun_test.go:389]: maybe include the remaining expected events too

fmt.Errorf("timed out waiting for %d more events: %#v", len(expected), expected)

Comment (sbwsg) [reply pkg/reconciler/taskrun/taskrun_test.go]: Good idea; done!

**Review by vdemeester — COMMENTED** [review]
/lgtm
/cc @afrittoli

**Review by afrittoli — COMMENTED** [review]
Thanks for this!
I wonder if we should do the same thing in the pipeline run controller tests, but in any case this PR looks.

**Review by afrittoli — COMMENTED** [review]
/approve

**Pull request merged**